docs/zero_knowledge.md

```diff
@@ -1,136 +1,76 @@
-in more detail:
-
-Say you recorded the above procedure on camera, so you can later convince 
-your very skeptical friends that I know a valid solution.
-On the recording they would see that every edge you chose indeed connects two 
-vertices of different color. But that won't convince them (they are very 
-skeptical) since you and I could have played false and agreed on the vertices 
-to be uncovered beforehand.
-
-All this means that with a Zero-Knowledge proof you would for example be able 
-to prove that you know a password without ever giving it to any potentially 
-malicious other party or eavesdropper.  
-Online you could prove that you are you to just one person and they would 
-not be able to prove that to anyone else.  
-And in the context of blockchain you could prove that a transaction is valid 
-without revealing any information about the transaction itself.
-
-## Zero-Knowledge protocols
-
-When first described Zero-Knowledge proofs were interactive, meaning the 
-prover and verifier need to interact, like in our example above.
-Later it was discovered that a shared reference string is enough to achieve 
-computational zero-knowledge without requiring an interaction between prover 
-and verifier.
-
-Over time different Zero-Knowledge protocols have been discovered. They have 
-beautiful names like *STARK* (Scalable Transparent ARgument of Knowledge), 
-*BulletProof*, *SNARG* (Succinct Non-interactive ARGuments) and *SNARK* 
-(Succinct Non-interactive ARgument of Knowledge).  
-I will not explain the different protocols in detail here.
-Instead, I will just give a brief overview of what those fancy abbreviation 
-actually mean so you have a staring point for the next internet research 
-rabbit hole.
-
-- Scalable:  
-  Both prover and verrifier running times are scalable.
-- Transparent:  
-  Doesn't require a setup-phase which uses a non-public random parameter like 
-  the shared reference string mentioned above.
-- ARgument:  
-  Strictly mathematically speaking, many of the above protocols are no proofs 
-  but arguments. However in practice we can still think of them as a proofs.
-- Knowledge:  
-  The prover proofs that they themselves know the solution. In our above 
-  example that means that the prover doesn't prove that a solution exists 
-  but that they actually know the solution.
-- Succinct:  
-  The proofs are relatively small and easy to verify
-- Non-interactive:  
-  No interaction between prover and verifier is needed
+Conventional proof systems make sure that the prover cannot act maliciously. ZKP systems take this a step further, and also make sure the 
+verifier cannot act maliciously. After computing and sending a ZKP, the verifier (or an eavesdropper) only learns that the prover is in possession of $w$, nothing more. This fact has the following implications:
+
+- The verifier is not able to prove to a third party that he knows $w$ (because he does not know it). 
```

Comment:
For consistency: use $V$ for verifier and I want to try to use only gender neutral pronouns; so in this case they/them